pre release

humanshield89 opened the pull request:
This is the pre release commit 

## What is new ? 
* Added jackpot contract 
* Change the refuel ratio to 0.1%
* Each refuel burns half and sends half to jackpot where it will be added to the milestone jackpot 
* The top fuel providers for each milestone get the jackpot 
* 25% to the first place , 15% to the second , 10 % to the third and 50% gets divided to the rest of the winners 
* Users can only claim if they have a nickname (the user can still win but only after he sets a nck name that he could claim his winnings)
* Added direct burn to volume (could be used by the future contracts we create to burn when needed ) 
* Added `directFuelFor` where a user or contract can directly fuel and gives credit of the fuel to a different user can be very useful for our future contracts 
* Added freeloader (do not pay fuel but can not win or participate ) , fuelCreditors are contracts that pays fuel on transactions but credits the fuel to the receiver this will be useful for LPpools and other contracts 
* Unit test now cover all functionalities except the LP creation 
* Removed fuel piles 
* Can create infinite milestones 
* Before the takeoffBlock the Volume token behave as any other erc20 token and no fuel is paid
* takeOffBlock needs to be set in order to start the fuel and other volume functionalities , take off block can be any block in the future but can only be set once

humanshield89 committed:
pre release

humanshield89 commented:
@NikhilManikala  @LunarAI  Did you guys actually pull and run tests before approving ?

LunarAI commented:
> @NikhilManikala @LunarAI Did you guys actually pull and run tests before approving ?

Yes, why do you ask?

humanshield89 committed:
track fuel added per milestone

humanshield89 committed:
burn any leftover amount in allocations after we crash

humanshield89 committed:
adjust fuel efficiency

humanshield89 committed:
change the fuel calculation for tests to meets the new changes

humanshield89 committed:
This is more of a QOL changes
    1 . Better Logic for slippage calculation
    2 . Fix Naming to avoid confusion
    3 . Follow formatting and naming conventions
    4 . Add BSC Test net to trufle configs
    5 . Restructure the Contracts (Interface => Contracts)
    6 . Remove spelling mistakes and typos
    7 . Bump openzeppelin >4.3.1

humanshield89 committed:
Add A faucet for test networks

humanshield89 committed:
Better Migration logic
